sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/samples/TimeSeriesInsightsClientSample/QuerySamples.cs

```diff
@@ -74,15 +76,99 @@ await foreach (TimeSeriesPoint point in humidityEventsQueryAnalyzer.GetResultsAs
+        {
+            // In this sample, we demonstrate querying for series data with predefined numeric variables in a predefined Time Series type
+
+            // Setup
+
+            // First create the Time Series type along with the numeric variables
+            var timeSeriesTypes = new List<TimeSeriesType>();
+
+            var celsiusVariable = new NumericVariable(
+                new TimeSeriesExpression("$event.Temperature"),
+                new TimeSeriesExpression("avg($value)"));
+            var fahrenheitVariable = new NumericVariable(
+                new TimeSeriesExpression("$event.Temperature * 1.8 + 32"),
+                new TimeSeriesExpression("avg($value)"));
+
+            var celsiusVariableName = "TemperatureInCelsius";
+            var fahrenheitVariableName = "TemperatureInFahrenheit";
+            var variables = new Dictionary<string, TimeSeriesVariable>
+            {
+                { celsiusVariableName, celsiusVariable },
+                { fahrenheitVariableName, fahrenheitVariable }
+            };
+
+            timeSeriesTypes.Add(new TimeSeriesType("TemperatureSensor", variables) { Id = "TemperatureSensorTypeId" });
+
+            Response<TimeSeriesTypeOperationResult[]> createTypesResult = await client
+                .Types
+                .CreateOrReplaceAsync(timeSeriesTypes)
+                .ConfigureAwait(false);
+
+            if (createTypesResult.Value.First().Error != null)
+            {
+                Console.WriteLine($"\n\nFailed to create a Time Series Insights type. " +
+                    $"Error Message: '{createTypesResult.Value.First().Error.Message}.' " +
+                    $"Code: '{createTypesResult.Value.First().Error.Code}'.");
+            }
+            else
+            {
+                // Get the Time Series instance and replace its type with the one we just created
+                Response<InstancesOperationResult[]> getInstanceResult = await client.Instances.GetAsync(new List<TimeSeriesId> { tsId });
+                if (getInstanceResult.Value.First().Error != null)
+                {
+                    Console.WriteLine($"\n\nFailed to retrieve Time Series instance with Id '{tsId}'. " +
+                        $"Error Message: '{getInstanceResult.Value.First().Error.Message}.' " +
+                        $"Code: '{getInstanceResult.Value.First().Error.Code}'.");
+                }
+                else
+                {
+                    TimeSeriesInstance instanceToReplace = getInstanceResult.Value.First().Instance;
+                    instanceToReplace.TypeId = createTypesResult.Value.First().TimeSeriesType.Id;
+                    Response<InstancesOperationResult[]> replaceInstanceResult = await client.Instances.ReplaceAsync(new List<TimeSeriesInstance> { instanceToReplace });
+                    if (replaceInstanceResult.Value.First().Error != null)
```

Comment:
First because there is only one instance?

Reply:
Yup. I usually use loops and not assume 1 instance. But because this is part of setting up for the actual snippet to be demonstrated, I used First()